x-pack/plugins/kubernetes_security/public/components/count_widget/helpers.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CountResult } from '../../../common/types/count';
+import { DEFAULT_QUERY } from '../../../common/constants';
+import { QueryDslQueryContainerBool } from '../../types';
+
+export const addResourceTypeToFilterQuery = (
+  filterQuery: string | undefined,
+  resourceType: 'node' | 'pod'
+) => {
+  let validFilterQuery = DEFAULT_QUERY;
+
+  try {
+    const parsedFilterQuery: QueryDslQueryContainerBool = JSON.parse(filterQuery || '{}');
+    if (!(parsedFilterQuery?.bool?.filter && Array.isArray(parsedFilterQuery.bool.filter))) {
+      throw new Error('Invalid filter query');
+    }
+    parsedFilterQuery.bool.filter.push({
+      bool: {
+        should: [
+          {
+            match_phrase: {
+              'orchestrator.resource.type': resourceType,
+            },
+          },
+        ],
+      },
+    });
+    validFilterQuery = JSON.stringify(parsedFilterQuery);
+  } catch {
+    // no-op since validFilterQuery is initialized to be DEFAULT_QUERY
+  }
+
+  return validFilterQuery;
+};
+
+export const numberFormatter = (num: CountResult) => {
+  if (Number(num) < 1e3) {
+    return num.toString();
+  }
+  if (Number(num) < 1e6) {
+    const newNum = Math.floor(Number(num) / 1000) * 1000;
+    return new Intl.NumberFormat('en-GB', {
+      // @ts-ignore
+      notation: 'compact',
+      compactDisplay: 'short',
+    }).format(Number(newNum));
+  }
+  return new Intl.NumberFormat('en-GB', {
```

Comment:
it seems like the floor is only used for for < 1M, but in the below case we want to display 999M, it's currently being rounded instead
<img width="251" alt="image" src="https://user-images.githubusercontent.com/16872649/178361802-c1eb3e4f-c442-46aa-988c-2ab43b6807b4.png">